modules/security/pages/encrypting-connections.adoc

```diff
@@ -1,167 +1,230 @@
+Learn more: link:https://httpd.apache.org/docs/2.4/ssl/ssl_intro.html[SSL/TLS Strong Encryption: An Introduction]
 
 == Nginx-Based
 
-TigerGraph uses the Nginx web server, so SSL configuration makes use of some built-in support in Nginx.
+TigerGraph uses the Nginx web server, so SSL/TLS configuration relies on Nginx’s built-in HTTPS support.
 
-http://nginx.org/en/docs/http/configuring_https_servers.html
+For additional reference, see the official Nginx documentation:
+link:http://nginx.org/en/docs/http/configuring_https_servers.html[Configuring HTTPS Servers (Nginx)].
 
-== Step 1. Obtain an SSL Certificate
+== Step 1: Obtain and Install Certificates
 
-The two main options for obtaining an SSL Certificate are to generate your own self-signed certificate or to purchase a certificate from a trusted Certificate Authority. Regardless of which method you choose, your certificate should be chained to a trusted root certificate embedded in your browser. The options and details for producing a trusted SSL certificate are beyond the scope of this document. The focus of this document is how to configure your TigerGraph system to use the certificate to enable SSL.
+You need certificates for your server (always) and for each client (only for mTLS).
 
-=== Option 1: Using a Certificate From A Trusted Agent
+=== Option A: Use a Certificate from a Trusted Authority
 
-First, obtain an SSL certificate from a trusted agent of your choice. Certificate vendors will provide clear instructions for ordering a certificate and then for installing it on your system.
+Obtain a certificate from a trusted Certificate Authority (CA).
+CA vendors provide guidance for requesting and installing certificates.
 
-Then you can configure the certificate with `gadmin config entry ssl`
+Configure in TigerGraph:
+[source,console]
+----
+gadmin config entry ssl
+----
 
-=== Option 2: Create a Self-Signed Certificate
+=== Option B: Create a Self-Signed Certificate
 
-There are multiple ways to create a self-signed certificate.  One example is shown below.
+For testing or internal use, you can generate your own certificate:
+
+[source,console]
+----
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout <PRIVATE_KEY_PATH> -out <CERTIFICATE_PATH>
+----
+
+Replace `<PRIVATE_KEY_PATH>` with the path to your private key and `<CERTIFICATE_PATH>` with the path to your public certificate.
+
+You will be prompted for certificate fields such as country, locality, organization, and hostname (Common Name).
 
 [WARNING]
 ====
-For simplicity, the method below will use the root certificate directly as the HTTPS server certificate.  This method is satisfactory for testing but should not be used for a production system.
+The self-signed process below uses the root certificate as the HTTPS server certificate. This is suitable for testing—not for production.
 ====
```

Comment:
The important part of this is the recommendation not to use this for production, but we already said that.